pkg/volumemanager/persistent_volume_manager.go

```diff
@@ -0,0 +1,154 @@
+}
+
+// NewPersistentVolumeManager creates a new PersistentVolumeManager
+func NewPersistentVolumeManager(kubeClient client.Interface) *PersistentVolumeManager {
+	volumeStore := NewPersistentVolumeOrderedIndex()
+	volumeReflector := cache.NewReflector(
+		&cache.ListWatch{
+			ListFunc: func() (runtime.Object, error) {
+				return kubeClient.PersistentVolumes().List(labels.Everything(), fields.Everything())
+			},
+			WatchFunc: func(resourceVersion string) (watch.Interface, error) {
+				return kubeClient.PersistentVolumes().Watch(labels.Everything(), fields.Everything(), resourceVersion)
+			},
+		},
+		&api.PersistentVolume{},
+		volumeStore,
+		0,
+	)
+	volumeReflector.Run()
+
+	claimStore := cache.NewStore(cache.MetaNamespaceKeyFunc)
+	claimReflector := cache.NewReflector(
+		&cache.ListWatch{
+			ListFunc: func() (runtime.Object, error) {
+				return kubeClient.PersistentVolumeClaims(api.NamespaceAll).List(labels.Everything(), fields.Everything())
+			},
+			WatchFunc: func(resourceVersion string) (watch.Interface, error) {
+				return kubeClient.PersistentVolumeClaims(api.NamespaceAll).Watch(labels.Everything(), fields.Everything(), resourceVersion)
+			},
+		},
+		&api.PersistentVolumeClaim{},
+		claimStore,
+		0,
+	)
+	claimReflector.Run()
+
+	manager := &PersistentVolumeManager{
+		volumeStore: volumeStore,
+		claimStore:  claimStore,
+		client:      kubeClient,
+	}
+	manager.syncHandler = manager.syncPersistentVolumeClaim
+
+	return manager
+}
+
+func (controller *PersistentVolumeManager) Run(period time.Duration) {
+	glog.V(5).Infof("Starting PersistentVolumeManager\n")
+	go util.Forever(func() { controller.synchronize() }, period)
+}
+
+func (controller *PersistentVolumeManager) synchronize() {
+	items := controller.claimStore.List()
+	wg := sync.WaitGroup{}
+	wg.Add(len(items))
+	for ix := range items {
+		go func(ix int) {
+			defer wg.Done()
+			claim := items[ix].(*api.PersistentVolumeClaim)
+			glog.V(5).Infof("Periodic sync of persistent volume claim: %v", claim.Name)
+			err := controller.syncHandler(claim)
+			if err != nil {
+				glog.Errorf("Error synchronizing: %v", err)
+			}
+		}(ix)
+	}
+	wg.Wait()
+}
+
+func (controller *PersistentVolumeManager) syncPersistentVolumeClaim(claim *api.PersistentVolumeClaim) error {
+	glog.V(5).Infof("Synchronizing persistent volume claim: %s\n", claim.Name)
+
+	if claim.Status.VolumeRef != nil {
+		glog.V(5).Infof("PersistentVolumeClaim[UID=%s] is bound to PersistentVolume[UID=%s]\n", claim.Name, claim.Status.VolumeRef.Name)
+		return nil
+	}
+
+	volume, err := controller.volumeStore.FindByClaim(claim)
+	if err != nil {
+		return err
+	}
+
+	if volume != nil {
+		claimRef, err := api.GetReference(claim)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
+		}
+
+		volumeRef, err := api.GetReference(volume)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting volume reference: %v\n", err)
+		}
+
+		volume.Spec.ClaimRef = claimRef
```

Comment:
Because @bgrant0607 said so :)

Anything persisted needs to be on Spec whereas Status is recreated by observation, even though my observers are persisting status to the new /status subresource.